synapse/config/emailconfig.py

```diff
@@ -294,8 +307,16 @@ def read_config(self, config, **kwargs):
                 if not os.path.isfile(p):
                     raise ConfigError("Unable to find email template file %s" % (p,))
 
+        self.email_subjects = {}  # type: Dict[str, str]
```

Comment:
This somewhat feels like it would benefit from being an attrs class (the hard-coded keys usually work nicely with that) -- but that might be annoying to do.

Reply:
Actually now that you mention it it's probably better to make it an attrs class.

Reply:
Done in https://github.com/matrix-org/synapse/pull/7846/commits/16812d9df8c98b2dba68fe9315036b95c56e8227